tests/integration/test_charm.py

```diff
@@ -17,33 +25,124 @@ async def juju_run(unit, cmd):
     return stdout
 
 
-async def test_build_and_deploy(ops_test: OpsTest):
+async def test_workload_online_default(ops_test: OpsTest, app: Application):
     assert ops_test.model
-    charm = await ops_test.build_charm(".")
-    await ops_test.model.deploy("postgresql-k8s")
-    await ops_test.model.deploy(charm)
-    await ops_test.model.wait_for_idle(raise_on_error=False)
-    await ops_test.model.add_relation(
-        "postgresql-k8s:db",
-        "mattermost-k8s",
+    mmost_unit = app.units[0]
+    action = await mmost_unit.run("unit-get private-address")
+    curl_output = await juju_run(
```

Comment:
Can't you run a simple "request" ? Why calling curl in the container ?

The question si valid for all other "curl" made in the container

Reply:
I reworked this with requests